global_costmap:
  global_costmap:
    ros__parameters:
      update_frequency: 1.0
      publish_frequency: 1.0
      global_frame: map
      robot_base_frame: base_link
      use_sim_time: false
      # offset: front / back from base_link
      footprint: '[[0.488,-0.293],[0.488,0.293],[-0.412,0.293],[-0.412,-0.293]]'
      footprint_padding: 0.01
      track_unknown_space: true
      plugins: ["static_layer", "inflation_layer"] # "obstacle_layer", 
      # obstacle_layer:
      #   plugin: "nav2_costmap_2d::ObstacleLayer"
      #   enabled: True
      #   observation_sources: scan
      #   footprint_clearing_enabled: true
      #   max_obstacle_height: 2.0
      #   combination_method: 1
      #   scan:
      #     topic: /scan
      #     obstacle_max_range: 2.5
      #     obstacle_min_range: 0.0
      #     raytrace_max_range: 3.0
      #     raytrace_min_range: 0.0
      #     max_obstacle_height: 2.0
      #     min_obstacle_height: 0.0
      #     clearing: True
      #     marking: True
      #     data_type: "LaserScan"
      #     inf_is_valid: false
      static_layer:
        plugin: "nav2_costmap_2d::StaticLayer"
        map_subscribe_transient_local: True
      inflation_layer:
        plugin: "nav2_costmap_2d::InflationLayer"
        enabled: true
        inflation_radius: 2.0
        cost_scaling_factor: 3.0
        inflate_unknown: false
        inflate_around_unknown: true
      always_send_full_costmap: True
  global_costmap_client:
    ros__parameters:
      use_sim_time: false
  global_costmap_rclcpp_node:
    ros__parameters:
      use_sim_time: false

planner_server:
  ros__parameters:
    expected_planner_frequency: 20.0
    use_sim_time: True
    planner_plugins: ["GridBased"]
    GridBased:
      plugin: "nav2_navfn_planner/NavfnPlanner"
      tolerance: 0.5
      use_astar: true
      allow_unknown: true

planner_server_rclcpp_node:
  ros__parameters:
    use_sim_time: True
